modules/c1xBidAdapter.js

```diff
@@ -0,0 +1,207 @@
+    bidRequest.push({
+      method: 'GET',
+      url: URL,
+      data: payloadString,
+      bids: bidIdTags
+    })
+    return bidRequest;
+  },
+
+  interpretResponse: function(serverResponse, requests) {
+    serverResponse = serverResponse.body;
+    requests = requests.bids || [];
+    const currency = 'USD';
+    const bidResponses = [];
+    let netRevenue = false;
+
+    if (!serverResponse || serverResponse.error) {
+      let errorMessage = serverResponse.error;
+      logError(LOG_MSG.invalidBid + errorMessage);
+      return bidResponses;
+    } else {
+      serverResponse.forEach(bid => {
+        logInfo(bid)
+        if (bid.bid) {
+          if (bid.bidType === 'NET_BID') {
+            netRevenue = !netRevenue;
+          }
+          const curBid = {
+            requestId: bid.bidId,
+            width: bid.width,
+            height: bid.height,
+            cpm: bid.cpm,
+            ad: bid.ad,
+            creativeId: bid.crid,
+            currency: currency,
+            ttl: 300,
+            netRevenue: netRevenue
+          };
+
+          if (bid.dealId) {
+            curBid['dealId'] = bid.dealId
+          }
+
+          for (let i = 0; i < requests.length; i++) {
+            if (bid.adId === requests[i].adUnitCode) {
+              curBid.requestId = requests[i].bidId;
+            }
+          }
+          logInfo(LOG_MSG.bidWin + bid.adId + ' size: ' + curBid.width + 'x' + curBid.height);
+          bidResponses.push(curBid);
+        } else {
+          // no bid
+          logInfo(LOG_MSG.noBid + bid.adId);
+        }
+      });
+    }
+
+    return bidResponses;
+  }
+
+}
+
+function bidToTag(bid, index) {
+  const tag = {};
+  const adIndex = 'a' + (index + 1).toString(); // ad unit id for c1x
+  const sizeKey = adIndex + 's';
+  const priceKey = adIndex + 'p';
+  const dealKey = adIndex + 'd';
+  // TODO: Multiple Floor Prices
+
+  const sizesArr = bid.sizes;
+  const floorPriceMap = bid.params.floorPriceMap || '';
+
+  const dealId = bid.params.dealId || '';
+
+  if (dealId) {
+    tag[dealKey] = dealId;
+  }
+
+  tag[adIndex] = bid.adUnitCode;
+  tag[sizeKey] = sizesArr.reduce((prev, current) => prev + (prev === '' ? '' : ',') + current.join('x'), '');
+  const newSizeArr = tag[sizeKey].split(',');
+  if (floorPriceMap) {
+    newSizeArr.forEach(size => {
+      if (size in floorPriceMap) {
+        tag[priceKey] = floorPriceMap[size].toString();
+      } // we only accept one cpm price in floorPriceMap
+    });
+  }
+  if (bid.params.pageurl) {
+    tag['pageurl'] = bid.params.pageurl;
+  }
+
+  return tag;
+}
+
+function bidToShortTag(bid) {
+  const tag = {};
+  tag.adUnitCode = bid.adUnitCode;
+  tag.bidId = bid.bidId;
+  return tag;
+}
+
+function stringifyPayload(payload) {
+  let payloadString = [];
+  for (var key in payload) {
+    if (payload.hasOwnProperty(key)) {
+      payloadString.push(key + '=' + payload[key]);
+    }
+  }
+  return payloadString.join('&');
```

Comment:
This function seems to have dropped string validation, is that intentional?

```
	function stringifyPayload(payload) {	
  let payloadString = '';	
  payloadString = JSON.stringify(payload).replace(/":"|","|{"|"}/g, (foundChar) => {	
    if (foundChar == '":"') return '=';	
    else if (foundChar == '","') return '&';	
    else return '';	
  });	
  return payloadString;	
}
```

Reply:
Could this be reintroduced with tests written to support cases it is looking to prevent

Reply:
yes, it was intentional, we have faced some problems while creating the request from JSON so we used the current code to overcome that issue. cc @vishnu1211